sdk/storage/storage-blob/samples/typescript/proxyAuth.ts

```diff
@@ -0,0 +1,41 @@
+/* 
+ Setup: Enter your storage account name and shared key in main()
+*/
+
+import { BlobServiceClient, StorageClient, SharedKeyCredential } from "../../src"; // Change to "@azure/storage-blob" in your package
+
+async function main() {
+  // Enter your storage account name and shared key
+  const account = "";
+  const accountKey = "";
+
+  // Use SharedKeyCredential with storage account and account key
+  const sharedKeyCredential = new SharedKeyCredential(account, accountKey);
+
+  // Use sharedKeyCredential, tokenCredential or anonymousCredential to create a pipeline
+  const pipeline = StorageClient.newPipeline(sharedKeyCredential, {
+    proxyOptions: { proxySettings: "http://localhost:3128" }
```

Comment:
If `https` is used instead of `http`.

The test fails with this error.
>tunneling socket could not be established, cause=write EPROTO 6156:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:openssl\ssl\record\ssl3_record.c:252:


cc: @bterlson

Reply:
It's likely an issue in the underlying axios library.  

Harsha already verified that `http` works fine.  Maybe we can merge this PR and track the external `https` issue separately.

Reply:
https://github.com/axios/axios/issues/1877

Reply:
Are we sure this is an Axios problem and not the proxy server not supporting HTTPS? HTTPS on proxies is hard to set up as you need a trusted root certificate to man-in-the-middle the connection without generating certificate validation errors.

Reply:
I ran a squid container on my azure VM and tested that the proxy server is working using `request` library

```javascript
var request = require('request');

request({
  'url':'https://docs.microsoft.com/en-us/rest/api/securitycenter/',
  'method': "GET",
  'proxy':'http://<server-ip>:3128'
},function (error, response, body) {
  if (!error && response.statusCode == 200) {
    console.log(body);
  } else {
    console.log("Error!!!" + error);
  }
})
```

Reply:
The proxy there is HTTP, however, so you're talking HTTP to the proxy and the proxy is talking HTTPS to the endpoint. Try with `https` for proxy.

Reply:
ah, proxy with `https` gave the same error 
```
Error: tunneling socket could not be established, cause=write EPROTO 41488:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:c:\ws\deps\openssl\openssl\ssl\record\ssl3_record.c:252:
```

Reply:
https://github.com/Azure/azure-sdk-for-js/issues/3465